beacon-chain/rpc/nodev1/node.go

```diff
@@ -9,14 +9,53 @@ import (
 	ptypes "github.com/gogo/protobuf/types"
 	"github.com/pkg/errors"
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
 	"github.com/prysmaticlabs/prysm/shared/version"
 	"go.opencensus.io/trace"
 	"google.golang.org/grpc/status"
 )
 
 // GetIdentity retrieves data about the node's network presence.
 func (ns *Server) GetIdentity(ctx context.Context, _ *ptypes.Empty) (*ethpb.IdentityResponse, error) {
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "nodeV1.GetIdentity")
+	defer span.End()
+
+	peerId := ns.PeerManager.PeerID().Pretty()
+
+	serializedEnr, err := p2p.SerializeENR(ns.PeerManager.ENR())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not obtain enr")
+	}
+	enr := fmt.Sprint("enr:", serializedEnr)
+
+	var p2pAddresses []string
+	for _, address := range ns.PeerManager.Host().Addrs() {
+		p2pAddresses = append(p2pAddresses, fmt.Sprint(address.String(), "/p2p/", peerId))
+	}
+
+	disc, err := ns.PeerManager.DiscoveryAddresses()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not obtain discovery address")
+	}
+	var discoveryAddresses []string
```

Comment:
```suggestion
	discoveryAddresses := make([]string, len(disc))
```
and then do not use append but rather assignment at index